cmd/promtool/tsdb.go

```diff
@@ -692,3 +736,39 @@ func backfillOpenMetrics(path, outputDir string, humanReadable, quiet bool, maxB
 
 	return checkErr(backfill(5000, inputFile.Bytes(), outputDir, humanReadable, quiet, maxBlockDuration))
 }
+
+func displayHistogram(dataType string, datas []int, total int) {
+	slices.Sort(datas)
+	start, end, step := generateBucket(datas[0], datas[len(datas)-1])
+	sum := 0
+	buckets := make([]int, (end-start)/step+1)
+	for _, c := range datas {
+		sum += c
+		buckets[(c-start)/step]++
+	}
+	avg := sum / len(datas)
+	fmt.Printf("%s (min/max/avg): %d/%d/%d\n", dataType, datas[0], datas[len(datas)-1], avg)
+	for bucket, count := range buckets {
+		percentage := 100.0 * count / total
+		fmt.Printf("[%d, %d]: %d ", bucket*step+start+1, (bucket+1)*step+start, count)
```

Comment:
If outside the loop it does something like:

```
maxValue := (buckets[len(buckets)-1]+1)*step+start
maxLen := strconv.Itoa(len(fmt.Sprintf("%d", maxValue)))
```

Then it can use a format inside it that will make things align and look nice, something like:

`"... % " + maxLen + "d ..."`

(Probably better to do it ourselves than pull in a package just for this.)

Reply:
since the variable `end` stores the max value to display, so I directly use `end` as `maxValue`

now the output is aligned like following

```
Size of float chunks
bytes per float chunk (min/max/avg): 121/1183/283
[   1, 1000]: 66 ######################################
[1001, 2000]: 4 ##
[2001, 3000]: 0
```